src/components/ClientCard/ClientCard.jsx

```diff
@@ -4,38 +4,107 @@
   * TODO: style based on currently selected card.
 */
 
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import moment from 'moment';
 import { connect } from 'react-redux';
+
 import { setActiveChat, loadMessages } from '../../store/Chat';
+
 import './ClientCard.css';
 
-const ClientCard = props => (
-  <li className="ClientCard" >
-    <button
-      className="ClientCard__btn"
-      onClick={() => {
-        props.setActiveChat(props.chat);
-
-        props.loadMessages(props.config, props.chat.id);
-      }}
-    >
-      {props.children}
-    </button>
-  </li>
-);
-
-ClientCard.propTypes = {
-  chat: PropTypes.object.isRequired,
-  config: PropTypes.object.isRequired,
-  children: PropTypes.node.isRequired,
-  setActiveChat: PropTypes.func.isRequired,
-  loadMessages: PropTypes.func.isRequired,
-};
+// TODO: Improve this by pulling ping/pong from daemon's connection with the
+//   client.
+// If a client is not active for 2 minutes, consider them offline
+const ONLINE_TIMEOUT = 120000;
+
+class ClientCard extends Component {
+  static propTypes = {
+    chat: PropTypes.object.isRequired,
+    messages: PropTypes.array.isRequired,
+    config: PropTypes.object.isRequired,
+    children: PropTypes.node.isRequired,
+    setActiveChat: PropTypes.func.isRequired,
+    loadMessages: PropTypes.func.isRequired,
+    activeId: PropTypes.string,
+  }
+
+  getChatLastMessage () {
+    const { messages, chat, activeId } = this.props;
+    let filteredMessages = messages.filter(msg => (
+      msg.chat === chat.id && msg.author === `client.${msg.chat}`
+    ));
+
+    if (filteredMessages.length > 0) {
+      return filteredMessages[filteredMessages.length - 1];
+    }
+
+    return false;
+  }
+
+  select () {
+
+  }
+
+  render () {
+    const { chat, config, activeId } = this.props;
+    const lastMessage = this.getChatLastMessage();
+    const classes = [
+      'ClientCard',
+      (activeId === chat.id) ? 'active' : '',
+    ];
+    const statusClasses = [
+      'ClientCard__status',
+      (lastMessage.timestamp &&
+        (new Date(lastMessage.timestamp) >= new Date().getTime() - ONLINE_TIMEOUT)) ?
+      'online' :
+      'offline',
+    ];
+
+    return (
+      <li className={classes.join(' ')}>
+        <button
+          onClick={() => {
+            this.props.setActiveChat(chat);
 
+            this.props.loadMessages(config, chat.id);
+          }}
+          className="ClientCard__btn"
+        >
+          <div className="ClientCard__icon ss-icon">
+            <span>user</span>
+          </div>
+          <div className="ClientCard__information">
+            <div className="ClientCard__information-row">
+              <span className="ClientCard__name">{this.props.children}</span>
+              <span className={statusClasses.join(' ')} />
+            </div>
+            <div className="ClientCard__information-row">
```

Comment:
thoughts on removing ternaries out of the `return` ? Bronek and I are pretty on the same page that we aim to keep our return statement to pure view syntax. Well, at least that's the way I am now. I find it a  bit hard to read.

Reply:
Yeah agreed, ill update

Reply:
thank youuuuu

Reply:
Fixed